src/coreclr/jit/lsraxarch.cpp

```diff
@@ -2488,6 +2488,52 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
                 break;
             }
 
+            case NI_AVX512F_BlendVariableMask:
+            {
+                assert(numArgs == 3);
+
+                if (op2->IsEmbMaskOp())
+                {
+                    // TODO-AVX512-CQ: Ensure we can support embedded operations on RMW intrinsics
+                    assert(!op2->isRMWHWIntrinsic(compiler));
```

Comment:
This will need some more handling for cases like `FMA` which are RMW already. Those cases may need to emit a `movaps` with embedded masking + the underlying instruction with embedded masking to ensure we get all the correct codegen.